### PR TITLE
Make space available as dropdown hotkey & only trigger hotkey when not filtered

### PR DESCRIPTION
**Pull request by jmoerschbach**
Re  #5407

### CODE REVIEW

**Comment by jmoerschbach** [comment]
hmm, false positive? Without this include it does not compile...

**Comment by jmoerschbach** [comment]
Hmm, I think the find_unused_includes script may have an error.
When a header file is scanned (here: dropdown.h), we look into each header file that is included from the header file to be checked and search for class definitions but not for function definitions.
https://github.com/widelands/widelands/blob/45e168b317f6f025f604f16d8476d758930f72f3/utils/find_unused_includes.py#L228

Changing the boolean param from `False` to `True` fixes the problem but I am not sure if this is smart...?

**Comment by Noordfrees** [comment]
Would perhaps adding the file to `DIFFICULT_FILES` (line 42) solve this?

**Comment by jmoerschbach** [comment]
> Would perhaps adding the file to `DIFFICULT_FILES` (line 42) solve this?

yes, of course :) way better :+1:

File: src/ui_basic/dropdown.h [hunk 1]
```diff
@@ -405,17 +406,32 @@ template <typename Entry> class Dropdown : public BaseDropdown {
...
 	bool check_hotkey_match(const std::string& input_text) {
 		for (auto& x : unfiltered_entries) {
-			if (input_text == to_lower(x.hotkey) && is_in_filtered_list(x.value)) {
-				if (hotkey_fn_) {
-					hotkey_fn_(x.value);
-				} else {
-					verb_log_dbg("hotkey match: %s but no hotkey function available!", x.hotkey.c_str());
+			if (" " == input_text) {
+				SDL_Keycode c = SDLK_SPACE;
+				const std::string& localized_space =
```
Comment (Noordfrees) [inline src/ui_basic/dropdown.h]:
```suggestion
				const std::string localized_space =
```
`to_lower` does not return a reference

**Review by Noordfrees — COMMENTED** [review]
One comment inline, but I'm not sure what this is supposed to do actually. <kbd>Space</kbd> is a reserved button, so it cannot be assigned as a hotkey at all. Selecting the highlighted entry with Space is still not working.

**Comment by jmoerschbach** [comment]
> One comment inline, but I'm not sure what this is supposed to do actually. Space is a reserved button, so it cannot be assigned as a hotkey at all. Selecting the highlighted entry with Space is still not working.

there is a space hotkey set to toggle buildhelp in show/hide menu. On master, hitting space modifies the filter, on this branch the hotkey is matched, toggling the buildhelp and closing the dropdown, see screenshot (master):
spaceNoHotkey

If we want to select with space then this hotkey cannot be assigned to space (and nowhere else) imho (as said in https://github.com/widelands/widelands/issues/5407#issuecomment-1140499258)

**Comment by Noordfrees** [comment]
I forgot about that one. Fix confirmed. I noticed two bugs though: 
- Filtering the dropdown with `E` works fine, but filtering it with `O` closes the dropdown instead of filtering it, even though no entry has O as a hotkey. Similarly, filtering with only `D` or only `A` works fine but filtering for the text `DA` closes it.
  For reference here the translated dropdown:
  image

- Hotkeys are interpreted even while the dropdown is already being filtered, which seems strange to me. E.g. in the above dropdown typing `G` `E` `W` starts filtering first for `G`, then for `GE`, and then it selects the workarea overlap entry instead of appending to the filter.

**Comment by jmoerschbach** [comment]
> 

fixed the second issue.
The first issue is on master as well and is not related to this pr. Should be investigated separately

**Comment by jmoerschbach** [comment]
does anybody know what is wrong here?

**Comment by frankystone** [comment]
~~~~
The following packages have unmet dependencies:
 udev : Breaks: systemd (< 249.11-0ubuntu3.3) but 249.11-0ubuntu3.1 is to be installed
E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.
Error: Process completed with exit code 100.
~~~~

Looks like the repo is not in sync.  I try to rerun the failed jobs, maybe this helps.

**Comment by matthiakl** [comment]
> 
> Looks like the repo is not in sync. I try to rerun the failed jobs, maybe this helps.

See #5467

**Review by Noordfrees — APPROVED** [review]

**Pull request merged**